### PR TITLE
use -Woverloaded-virtual with g++.  quiet compiler warnings

### PR DESCRIPTION
**Pull request by sethalves**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: b8cf4ea29de72a79201c05b4de19b3c509124926 available at: https://gist.github.com/86262c7580c1b0ffff5d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5141.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5141.dmg

File: libraries/entities-renderer/src/RenderablePolyVoxEntityItem.h [hunk 1]
```diff
@@ -45,6 +45,7 @@ class RenderablePolyVoxEntityItem : public PolyVoxEntityItem {
                          bool& keepSearching, OctreeElement*& element, float& distance, BoxFace& face, 
                          void** intersectedObject, bool precisionPicking) const;
 
+    using PolyVoxEntityItem::setVoxelSurfaceStyle;
```
Comment (ZappoMan) [inline libraries/entities-renderer/src/RenderablePolyVoxEntityItem.h]: what exactly does this do?

Comment (sethalves) [inline libraries/entities-renderer/src/RenderablePolyVoxEntityItem.h]: The problem appears to be caused by having a virtual function with a signature unlike that of a function with the same name in a base class.

http://stackoverflow.com/questions/4146499/why-does-a-virtual-function-get-hidden

Comment (ZappoMan) [inline libraries/entities-renderer/src/RenderablePolyVoxEntityItem.h]: there's a bunch of similar warnings in all the bullet headers anyway we can work around them?

Comment (ZappoMan) [inline libraries/entities-renderer/src/RenderablePolyVoxEntityItem.h]: but the problem is... when you do this, which method do you expect to be called when some code calls the virtual function with a pointer to the base class? The warning is telling you, you likely won't get the behavior you expected... in other words, you lose your virtualness.

Comment (sethalves) [inline libraries/entities-renderer/src/RenderablePolyVoxEntityItem.h]: I'm not seeing them.  Maybe if we merge this, I will!

**Comment by hifi-gustavo** [comment]
Build console output for commit: 22ac61e476ee43f1beabcd736314ad2ddab2051b available at: https://gist.github.com/86262c7580c1b0ffff5d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5141.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5141.dmg

File: libraries/entities/src/PolyVoxEntityItem.h [hunk 2]
```diff
@@ -61,10 +61,14 @@ class PolyVoxEntityItem : public EntityItem {
...
 
-    virtual void setVoxelSurfaceStyle(PolyVoxSurfaceStyle voxelSurfaceStyle) { _voxelSurfaceStyle = voxelSurfaceStyle; }
-    virtual void setVoxelSurfaceStyle(uint16_t voxelSurfaceStyle) {
-        setVoxelSurfaceStyle((PolyVoxSurfaceStyle) voxelSurfaceStyle);
+    void setVoxelSurfaceStyle(PolyVoxSurfaceStyle voxelSurfaceStyle) {
+        if (voxelSurfaceStyle == _voxelSurfaceStyle) {
+            return;
+        }
+        updateVoxelSurfaceStyle(voxelSurfaceStyle);
     }
```
Comment (AndrewMeadows) [inline libraries/entities/src/PolyVoxEntityItem.h]: Coding standard section **2.1.3** says "Most function implementations should reside in the source file".  In particular multi-line functions need to be in the cpp or file or declared in the class and implemented as inline below.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7d707447ca749ed7a3aef93b3d2b00666bcdce77 available at: https://gist.github.com/86262c7580c1b0ffff5d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5141.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5141.dmg

**Pull request merged**